Shared/mods/deathmatch/logic/lua/CLuaFunctionParser.h

```diff
@@ -541,6 +551,19 @@ struct CLuaFunctionParserBase
             }
             return static_cast<T>(result);
         }
+        else if constexpr (std::is_same_v<T, CLuaArgument>)
+        {
+            int iType = lua_type(L, index);
+            if (iType != LUA_TNONE)
+            {
+                CLuaArgument argument;
+                argument.Read(L, index++);
+                return argument;
+            }
+
+            SetBadArgumentError(L, "value", index, "none");
```

Comment:
```suggestion
            SetBadArgumentError(L, "value", index - 1, "none");
```

Possible off-by-one error here? `argument.Read(L, index++);` above changes the index, so now you're pointing to the next argument

Reply:
As a result, this bad argument error also cannot be executed (since that'd be handled by `TypeMatch` already